Barotrauma/BarotraumaShared/SharedSource/Events/Missions/GoToMission.cs

```diff
@@ -4,9 +4,21 @@ namespace Barotrauma
 {
     partial class GoToMission : Mission
     {
+        private readonly bool stateControlsCompletion;
```

Comment:
To me this field seems unnecessary, this could work the same way as displayAsFailedState (= -1 means the value does nothing).

Reply:
Good point.